src/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -414,6 +415,37 @@ protected void paintComponent(final Graphics g) {
     updateStep();
     uiContext.addShutdownWindow(this);
   }
+  
+  
+  public static KeyListener getFlagToggleKeyListener(TripleAFrame frame) {
+    return new KeyListener() {
+      private boolean blockInputs = false;
+      @Override
+      public void keyTyped(final KeyEvent e) {/*Do nothing*/}
+
+      @Override
+      public void keyPressed(final KeyEvent e) {
+        if(!blockInputs){
```

Comment:
It looks like the flags will be on only as long as you hold the key down. I was thinking the hotkey would toggle it on / off, but this is okay too.